2016/submission/ref-util.pod

```diff
@@ -18,49 +30,129 @@ maintain: Checking the validity of references using the C<ref> function.
+by character with the C<HASH> string was passed in in the code.  That, of
+course, all happens really quickly, but it's more work (and more Perl virtual
+ops) than need happen just to tell if this was a hashref or not!
+
+So the code review had found two problems:  One, an abstraction was needed to
+make it more reliable.  Two, the code needed to involve less stuff to make it
+faster.
+
+=head2 The Voice of an Angel
+
+Thank goodness the Wise Old Elf read a L<CPAN Weekly|http://cpan-weekly.org/>
+email about this very pattern! Because of this he decided to try out a new
+module, L<Ref::Util>.
+
+L<Ref::Util> provides a set of helpful functions to determine what kind of
+reference a variable is. It abstracts several awkward ref checking patterns into
+new function.  So, as our above example:
 
     ref $args eq 'HASH'
         or croak('Sled expects arguments as a hashref');
 
-    # after
+Can be written as:
+
     use Ref::Util qw<is_plain_hash>;
 
     is_plain_hash($args)
         or croak('Sled expects arguments as a hashref');
 
 Or maybe we want to check for a blessed array reference, making sure it
-isn't accidentally blessed:
-
-    use Carp qw<croak>;
+isn't accidentally blessed? The following:
 
-    # before
     use Scalar::Util qw<blessed reftype>;
 
     blessed $args && ref $args && reftype($args) eq 'ARRAY'
         or Carp::croak('Uh oh, we require a blessed reference');
 
-    # after
+Can much more succinctly be written with the new C<is_blessed_arrayref>
+function:
+
     use Ref::Util qw<is_blessed_arrayref>;
 
     is_blessed_arrayref($args)
         or Carp::croak('Uh oh, we require a blessed reference');
 
 Not writing a lot of code but still being accurate is something Santa
-appreciates, but that's not the best part.
-
-These functions get compiled using custom op codes into a single op in
-the tree. While Santa is not exactly sure what this means, an elf
-helpfully explained that "it makes the sled go faster." And we don't
-even need to put racing stripes on the sled!
+appreciates, but that's not the best part....the best part is the speed
+increase.
+
+Remember when the Elves were worried about the overhead of introducing a new
+function call?  Well, Ref::Util doesn't do that.  It introduces a set of new
+custom ops to do the hard work instead which is much much faster than any kind
+of function call.
+
+To understand what that means requires an understanding of the internal of perl.
+When Perl code is run the code is first compiled into a series of "ops" -
+operations like "add these values", "push to list", or even as complicated as
+"run this regular expression."  These ops are somewhat like the machine code
+that runs on your actual processor, but much higher level and more complicated.
+When perl actually runs your code it runs a sort of virtual machine that
+basically looks at each of these operations in turn and does whatever operation
```

Comment:
This can possibly be simplified by saying "perl runs these ops" instead of creating a virtual machine. perl itself *is* the virtual machine. So, it becomes trickier to explain. Maybe just reducing complexity by reducing the explanation?s